compileopts/config.go

```diff
@@ -91,6 +99,12 @@ func (c *Config) BuildTags() []string {
 	for i := 1; i <= c.GoMinorVersion; i++ {
 		tags = append(tags, fmt.Sprintf("go1.%d", i))
 	}
+	if c.BuildMode() == "c-shared" && strings.HasPrefix(c.Triple(), "wasm") {
+		// Use reactor mode, which means the application is initialized in
+		// _initialize and main.main is not called by default. See for example:
+		// https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md
+		tags = append(tags, "tinygo_wasm_reactor")
```

Comment:
Proposal: rename `tinygo_wasm_reactor` to `tinygo.buildmode.c_shared`

Reply:
I guess we can do that, but `tinygo_wasm_reactor` seemed a bit more explicit. We can always change it later (I hope, if no external packages suddenly start depending on the build tag).

Reply:
Context: I think mapping the tag to the flag eases the developer experience, both for TinyGo developers and end users.

Reply:
I have fixed this by removing the tag altogether and selecting between entry points inside the compiler. (This is similar to how big Go does it).